enhancements/local-storage/automatic-detection.md

```diff
@@ -0,0 +1,349 @@
+	// + optional
+	VolumeMode PersistentVolumeMode `json:"volumeMode,omitempty"`
+	// FSType type to create when volumeMode is Filesystem
+	// +optional
+	FSType string `json:"fsType,omitempty"`
+	// If specified, a list of tolerations to pass to the discovery daemons.
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+	// DeviceInclusionSpec is the filtration rule for including a device in the device discovery
+	// +optional
+	DeviceInclusionSpec *DeviceInclusionSpec `json:"deviceInclusionSpec"`
+}
+
+// DeviceMechanicalProperty holds the device's mechanical spec. It can be rotational or nonRotational
+type DeviceMechanicalProperty string
+
+const (
+	// The mechanical properties of the devices
+	// Rotational refers to magnetic disks
+	Rotational DeviceMechanicalProperty = "Rotational"
+	// NonRotational refers to ssds
+	NonRotational DeviceMechanicalProperty = "NonRotational"
+)
+
+type DeviceType string
+
+const (
+	// The DeviceTypes that will be supported by the LSO.
+	// These Discovery policies will based on lsblk's type output
+	// Disk represents a device-type of disk
+	Disk DeviceType = "Disk"
+	// Part represents a device-type of partion
+	Partition DeviceType = "Part"
+)
+
+type DeviceTypeList struct {
+	// Devices is the list of devices that should be used for automatic detection.
+	// This would be one of the types supported by the local-storage operator. Currently,
+	// the supported types are: disk, part. If the list is empty the default value will be `[disk]`.
+	Devices []DeviceType `json:"devices"`
+}
+
+type DeviceInclusionSpec struct {
+	// DeviceTypeList holds the list of devices that should be used for automatic detection.
+	// If Nil the default detection will be of disks.
+	DeviceTypeList *DeviceTypeList `json:"deviceTypeList"`
+
+	// DeviceMechanicalProperty denotes whether Rotational or NonRotational disks should be used.
+	// by default, it selects both
+	// +optional
+	DeviceMechanicalProperty DeviceMechanicalProperty `json:"deviceMechanicalProperty"`
+
+	// MinSize is the minimum size of the device which needs to be included
+	// +optional
+	MinSize *resource.Quantity `json:"minSize"`
+
+	// MaxSize is the maximum size of the device which needs to be included
+	// +optional
+	MaxSize *resource.Quantity `json:"maxSize"`
+
+	// Models is a list of device models. If not empty, the device's model as outputted by lsblk needs
+	// to contain at least one of these strings.
+	// +optional
+	Models []string `json:"models"`
+
+	// Vendors is a list of device vendors. If not empty, the device's model as outputted by lsblk needs
+	// to contain at least one of these strings.
+	// +optional
+	Vendors []string `json:"vendors"`
+}
+
+type LocalVolumeGroupPhase string
+
+const (
+	DiscoveringPhase LocalVolumeGroupPhase = "Discovering"
+	FailedPhase      LocalVolumeGroupPhase = "Failed"
+	DiscoveredPhase  Phase                 = "Discovered"
+	ProvisionedPhase LocalVolumeGroupPhase = "Provisioned"
```

Comment:
When would you mark the `LocalVolumeGroup` object as "Provisioned" ?

Reply:
I m thinking when we are done with symlinking the device path?

Reply:
On all the nodes? how would you keep track of it? IMO this phase does not make sense.

Reply:
As discussed, updated to `Failed` and `Running`.